packages/xstate-react/src/useMachine.ts

```diff
@@ -113,6 +114,10 @@ export function useMachine<
   Interpreter<TContext, any, TEvent, TTypestate>['send'],
   Interpreter<TContext, any, TEvent, TTypestate>
 ] {
+  const machine = useMemo(() => {
+    return typeof getMachine === 'function' ? getMachine() : getMachine;
+  }, []);
+
   if (process.env.NODE_ENV !== 'production') {
```

Comment:
Given that we won't have `machine` resolved at this point in time we should change this condition to:
```suggestion
  if (process.env.NODE_ENV !== 'production' && typeof getMachine !== 'function') {
```